### PR TITLE
exhaustive (yet very basic) printer test, code clean up

### PR DESCRIPTION
**Pull request by aviatesk**
- exhaustive printer test
- clean up unnecessary `kwargs`

### CODE REVIEW

File: src/codeview.jl
Comment (simeonschaub) [inline src/codeview.jl:84]: This probably shouldn't use `@assert` since IIUC, it is also meant to be called directly and there have been discussions about disabling `@assert` statements by default in the future.

File: src/codeview.jl
Comment (simeonschaub) [inline src/codeview.jl:81]:
```suggestion
    src::Union{CodeInfo,IRCode}, @nospecialize(rt), mi::Union{Nothing,MethodInstance}=nothing;
```

File: src/Cthulhu.jl
Comment (timholy) [inline src/Cthulhu.jl:215]: Where does `selected` get defined now?

File: src/Cthulhu.jl
Comment (simeonschaub) [inline src/Cthulhu.jl:213]: Is `verbose` the best name here? It feels more like not hiding code from you. Perhaps `hide_type_stable` would be clearer?

File: src/codeview.jl
Comment (timholy) [inline src/codeview.jl:140]: This is not anything this PR needs to address, but I just added a comment about `verbose` in #189. Do you agree that the sense seems flipped?

**Review by timholy — COMMENTED** [review]
A very nice cleanup, thanks!

How hard do you think it would be to add more than "it works" tests? We may be covered elsewhere, of course, but the terminal tests I've added in #190 do get to some of that. So I don't think it's urgent here, and having exhaustive coverage has real advantages.

**Review by simeonschaub — APPROVED** [review]
Thanks for doing this, I really like the added consistency and clarity! This is technically a breaking user-facing change though, are we fine with that for a minor version? Would be good to at least get another opinion on this.

Awesome to finally have some basic tests! I have put this off for a while, but it's great to at least be certain that these don't start throwing errors.

**Comment by timholy** [comment]
> This is technically a breaking user-facing change though

Which part? I missed that bit.

Agreed, though, that the `verbose` option is pretty confusingly-named. Changing that would, of course, be a breaking change, but we could make it a deprecation instead.

**Comment by simeonschaub** [comment]
> > This is technically a breaking user-facing change though
> 
> Which part? I missed that bit.

It changes the order of arguments to `cthulhu_code_typed`/`cthulhu_code_warntype` and makes some of the arguments kwargs instead.

**Comment by timholy** [comment]
Doesn't that only affect the non-exported `cthulhu_typed`/`cthulhu_warntype`? I don't see any changes to the `descend_*` API.

**Comment by timholy** [comment]
Once we get this merged, I have a fix for #94 ready to go (it will likely conflict with this PR so I'll wait until this merges), and then I think it's time for a new release. Addressing the UI issues in #189 really should be a minor bump, not a patch bump.

File: src/Cthulhu.jl
Comment (aviatesk) [reply src/Cthulhu.jl]: Good catch, it's just because I didn't run JET on this PR :P

File: src/codeview.jl
Comment (aviatesk) [reply src/codeview.jl]: Thanks ! Now I just remember I got the same pointer from you on Julia Base 😅 Will make sure to use explicit error.

**Comment by aviatesk** [comment]
> How hard do you think it would be to add more than "it works" tests?

I'd like to add more specific and individual tests in the future. For example, these test cases don't cover any cases when `src::CodeInfo`, while `cthulhu_typed` really changes its behavior based on the type of `src`.
I think we need to setup each specific test cases for them, and I'd like to gradually enrich our test suite.

**Comment by aviatesk** [comment]
> Addressing the UI issues in #189 really should be a minor bump, not a patch bump.

Yeah, I'm just fine with a minor bump.

**Comment by aviatesk** [comment]
> Is verbose the best name here? It feels more like not hiding code from you. Perhaps hide_type_stable would be clearer?
> Agreed, though, that the verbose option is pretty confusingly-named. Changing that would, of course, be a breaking change, but we could make it a deprecation instead.

yeah, we should rename this name. Can we just address this in a separate PR though ? Just for the sake of easier reviews.

**Comment by timholy** [comment]
Right, assuming I'm not wrong about https://github.com/JuliaDebug/Cthulhu.jl/pull/191#issuecomment-891761616, our next release can be a patch release. Then fixing some of the issues in #189 would be a minor release since it will have significant user-visible change.

**Comment by simeonschaub** [comment]
> Doesn't that only affect the non-exported `cthulhu_typed`/`cthulhu_warntype`? I don't see any changes to the `descend_*` API.

Yeah, I thought they were part of the public API as well, but this is probably fine then with the minor version bump.

>  I think we need to setup each specific test cases for them, and I'd like to gradually enrich our test suite.

Yeah, we could probably just write the outputs of different configurations to a file and always test against that. The only worry would be to keep that in sync with Base if the printing there changes slightly or the optimizer gets more aggressive, but that's probably doable.

**Comment by timholy** [comment]
I don't think any of the [changes](https://github.com/JuliaDebug/Cthulhu.jl/compare/v2.1.2...master) need a minor version bump, as they all seem to be bugfixes, testing, and cleanup (which is patch territory). But either is OK with me, minor bumps are generally not disruptive post-1.0.

**Pull request merged**
